src/config/graphical-session/graphics-drivers/nvidia.md

```diff
@@ -0,0 +1,79 @@
+# NVIDIA
+
+## nouveau (Open Source Driver)
+
+Install the Mesa DRI driver `mesa-dri` and optionally the Xorg driver
+`xf86-video-nouveau`.
```

Comment:
i'm not sure i understand this. This part is talking about using `nouveau`, but installing `xf86-video-nouveau` is optional? The `mesa-dri` package doesn't pull it in; does `mesa-dri` alone somehow provide support for Nvidia hardware?

Reply:
Yes. It's not perfect, but nouveau can provide a modesetting driver for Xorg.

Reply:
Perhaps they can expand on this here.

Reply:
@ericonr: Well, i understand what the `nouveau` driver is, but if one wants to use it, doesn't one *have* to install `xf86-video-nouveau`? Or does `mesa-dri` provide the `nouveau` driver, with `xf86-video-nouveau` providing additional functionality?

Reply:
At the moment, this sentence reads as implying that to use the `nouveau` driver, one must install `mesa-dri`, but that installing `xf86-video-nouveau` isn't strictly needed.

Reply:
'nouveau driver' is not a well defined concept, because it could refer to the kernel driver, the Mesa DRI driver or the Xorg DDX driver, see first paragraph [here](https://nouveau.freedesktop.org/wiki/).
Xorg can use two drivers, `nouveau_drv.so` (provided by `xf86-video-nouveau`) or `modesetting_drv.so` (provided by `xorg-server` itself). The former is the legacy one and can work without `mesa-dri` afaik (it could be missing some functionality), while the latter is merely a wrapper that relies entirely on the DRI driver. The legacy one might be more stable, while modesetting might be more featureful. We could ask `#nouveau` for better info